crates/fast_compute/src/layer.rs

```diff
@@ -190,14 +198,65 @@ impl<T: TowerFamily, P: PackedTop<T>> ComputeLayer<T::B128> for FastCpuLayer<T,
 	fn accumulate_kernels(
 		&self,
 		_exec: &mut Self::Exec,
-		_map: impl for<'a> Fn(
+		map: impl Sync
+		+ for<'a> Fn(
 			&'a mut Self::KernelExec,
 			usize,
 			Vec<KernelBuffer<'a, T::B128, Self::DevMem>>,
 		) -> Result<Vec<Self::KernelValue>, Error>,
-		_mem_maps: Vec<KernelMemMap<'_, T::B128, Self::DevMem>>,
+		mem_maps: Vec<KernelMemMap<'_, T::B128, Self::DevMem>>,
 	) -> Result<Vec<Self::OpValue>, Error> {
-		unimplemented!()
+		let log_chunks_range = KernelMemMap::log_chunks_range(&mem_maps)
+			.ok_or_else(|| Error::InputValidation("no chunks range found".to_string()))?;
+
+		// Choose the number of chnks based on the range and the number of threads available.
```

Comment:
nit: chnks -> chunks

There's a pre-commit hook to catch typos now. Please install that or start running it regularly. (Or add to CI :-) )